### PR TITLE
Update configuration.asciidoc

### PR DESCRIPTION
**Pull request by leahleahy**

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -182,16 +181,16 @@ If this setting is set in both the web.config file and as an environment variabl
 
 <<dynamic-configuration, image:./images/dynamic-config.svg[] >>
 
-A boolean specifying if the agent should be recording or not.
-When recording, the agent captures HTTP requests, tracks errors, and collects and sends metrics.
-When not recording, the agent works as a noop, not collecting data and not communicating with the APM server,
+A Boolean specifying if the {agent} should be recording or not.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:184]: Should we also add the word value here? Boolean value?

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -182,16 +181,16 @@ If this setting is set in both the web.config file and as an environment variabl
...
+When recording, the {agent} captures HTTP requests, tracks errors, and collects and sends metrics.
+When not recording, the {agent} works as a noop, where it does not collect data or communicate with the APM server,
 except for polling the central configuration endpoint.
-As this is a reversible switch, agent threads are not killed when inactivated, but they will be
+This is a reversible switch, so the {agent} threads are not killed when deactivated, but they will be
```
Comment (leahleahy) [inline docs/configuration.asciidoc:188]: Deactivated instead of inactivated? Does that still make sense?

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -236,7 +235,7 @@ WARNING: Setting `Enabled` to `false` influences the behavior of the <<public-ap
 This is used to keep all the errors and transactions of your service together
 and is the primary filter in the Elastic APM user interface.
 
-NOTE: The service name must conform to this regular expression: `^[a-zA-Z0-9 _-]+$`. In less regexy terms: Your service name must only contain characters from the ASCII alphabet, numbers, dashes, underscores and spaces. Characters in service name which don't match regular expression will be replaced by `_` symbol.
+NOTE: The service name must conform to this regular expression: `^[a-zA-Z0-9 _-]+$`. Your service name must only contain characters from the ASCII alphabet, numbers, dashes, underscores, and spaces. Characters in service name that don't match regular expression will be replaced by `_` symbol.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:238]: Is it okay to remove the "in less regexy terms" ?

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -277,7 +276,7 @@ NOTE: This feature requires APM Server versions >= 7.5
 
 A version string for the currently deployed version of the service. If you don’t
 version your deployments, the recommended value for this field is the commit identifier
-of the deployed revision, e.g. the output of git rev-parse HEAD.
+of the deployed revision, e.g. the output of git rev-parse HEAD.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:279]: Lines 277-280 didn't really make sense. Could/Should this be rewritten?

File: docs/configuration.asciidoc [hunk 5]
```diff
@@ -591,7 +588,7 @@ the trust store, such as a self-signed certificate.
 
 The maximal amount of time events are held in the queue until there is enough to send a batch.
 It's possible for a batch to contain less than <<config-max-batch-event-count,`MaxBatchEventCount`>> events
-if there are events that need to be sent out because they were held for too long.
+if there are events that need to be sent out because they were held for too long.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:591]: Should this be rephrased?

File: docs/configuration.asciidoc [hunk 6]
```diff
@@ -623,13 +620,13 @@ in that case agent uses the default value instead.
 [[config-max-batch-event-count]]
 ==== `MaxBatchEventCount` (added[1.1])
 
-The maximal number of events to send in a batch.
+The maximum number of events to send in a batch.
```
Comment (leahleahy) [inline docs/configuration.asciidoc:623]: Is maximum here still correct instead of maximal?

File: docs/configuration.asciidoc [hunk 7]
```diff
@@ -25,7 +25,7 @@ This feature is enabled in the Agent by default, with <<config-central-config>>.
 === Configuration on ASP.NET Core
 
 The `UseElasticApm()` extension method offers an overload to pass an `IConfiguration` instance to the APM Agent.
-To use this type of setup, which is typical in an ASP.NET Core application, your application's `Startup.cs` file should contain code similar to the following:
+To use this type of setup, which is typical in an ASP.NET Core application, your applications `Startup.cs` file should contain code similar to the following:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:28]:
```suggestion
To use this type of setup, which is typical in an ASP.NET Core application, your application's `Startup.cs` file should contain code similar to the following:
```

File: docs/configuration.asciidoc [hunk 8]
```diff
@@ -88,8 +88,8 @@ Here is a sample `appsettings.json` configuration file for a typical ASP.NET Cor
 <1> With ASP.NET Core, you must set `LogLevel` for the internal APM logger in the standard `Logging` section with the `Elastic.Apm` category name.
 <2> The configurations below `ElasticApm` are fetched by the agent if the corresponding `IConfiguration` is passed to the agent.
 
-In certain scenarios--like when you're not using ASP.NET Core--you won't activate the agent with the `UseElasticApm()` method.
-In this case, you can set the agent's log level with <<config-log-level,`ElasticApm:LogLevel`>>, as shown in the following `appsettings.json` file:
+In certain scenarios--like when you're not using ASP.NET Core, the agent won't be activated with the `UseElasticApm()` method.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:91]:
```suggestion
In certain scenarios--you won't activate the agent with the `UseElasticApm()` method.
```

File: docs/configuration.asciidoc [hunk 9]
```diff
@@ -88,8 +88,8 @@ Here is a sample `appsettings.json` configuration file for a typical ASP.NET Cor
...
+In this case, set the agents log level with <<config-log-level,`ElasticApm:LogLevel`>>, as shown in the following `appsettings.json` file:
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:92]:
```suggestion
In this case, set the agent log level with <<config-log-level,`ElasticApm:LogLevel`>>, as shown in the following `appsettings.json` file:
```

File: docs/configuration.asciidoc [hunk 10]
```diff
@@ -146,15 +145,15 @@ Below is a sample `Web.config` configuration file for a ASP.NET application.
 ----
 
 Additionally, on ASP.NET, you can implement your own configuration reader. To do this, implement the `IConfigurationReader` interface from the `Elastic.Apm.Config` namespace.
-Once implemented, you can make use of the <<config-full-framework-configuration-reader-type, `FullFrameworkConfigurationReaderType`>> setting.
+Once implemented, use the <<config-full-framework-configuration-reader-type, `FullFrameworkConfigurationReaderType`>> setting.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:148]:
```suggestion
Once implemented, you can use the <<config-full-framework-configuration-reader-type, `FullFrameworkConfigurationReaderType`>> setting.
```

File: docs/configuration.asciidoc (hunk 3 above)
Comment (bmorelli25) [inline docs/configuration.asciidoc:238]:
```suggestion
NOTE: The service name must conform to this regular expression: `^[a-zA-Z0-9 _-]+$`. In other words, your service name must only contain characters from the ASCII alphabet, numbers, dashes, underscores, and spaces. Characters in service names that don't match the regular expression will be replaced with the `_` symbol. 
```

File: docs/configuration.asciidoc (hunk 4 above)
Comment (bmorelli25) [inline docs/configuration.asciidoc:279]:
```suggestion
of the deployed revision, e.g. the output of `git rev-parse --verify HEAD`. 
```

File: docs/configuration.asciidoc [hunk 11]
```diff
@@ -344,15 +343,15 @@ NOTE: On ASP.NET Core application the agent uses https://docs.microsoft.com/en-u
...
+By default, the agent samples every transaction (e.g. a request to your service).
+To reduce overhead and storage requirements, set the sample rate to a value between 0.0 and 1.0.
 The agent will still record the overall time and result for unsampled transactions,
 but no context information, labels, or spans will be recorded.
 
 NOTE: When parsing the value for this option, the agent doesn't consider the current culture.
 It also expects that a period (`.`) is used to separate the integer and the fraction of a floating-point number.
 
-This setting can be changed after agent's start.
+This setting can be changed after the agent is started.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:354]:
```suggestion
This setting can be changed after the agent starts.
```

File: docs/configuration.asciidoc [hunk 12]
```diff
@@ -372,16 +371,16 @@ This setting can be changed after agent's start.
...
+This is helpful when a transaction creates a very high amount of spans,
 for example, thousands of SQL queries.
 Setting an upper limit helps prevent overloading the Agent and APM server in these edge cases.
 
 NOTE: A value of `0` means that spans will never be collected.
 Setting `-1` means that spans will never be dropped.
 The Agent will revert to the default value if the value is set below `-1`.
 
-This setting can be changed after agent's start.
+This setting can be changed after agent is started.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:383]:
```suggestion
This setting can be changed after agent starts.
```

File: docs/configuration.asciidoc [hunk 13]
```diff
@@ -422,21 +421,19 @@ the latest {apm-app-ref}/agent-configuration.html[APM Agent configuration].
...
 The wildcard, `*`, matches zero or more characters, and matching is case insensitive by default.
 Prepending an element with `(?-i)` makes the matching case sensitive.
 Examples: `/foo/*/bar/*/baz*`, `*foo*`.
 
-You should review the data captured by Elastic APM carefully to make sure it does not contain sensitive information.
-If you do find sensitive data in your Elasticsearch index,
-you should add an additional entry to this list.
-Make sure to include the default entries as well, as setting a value here will overwrite the defaults.
+Please be sure to review the data captured by Elastic APM carefully to make sure it does not contain sensitive information.
+If you do find sensitive data in your Elasticsearch index, add an additional entry to this list.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:433]:
```suggestion
If you do find sensitive data in your {es} index, add an additional entry to this list.
```

File: docs/configuration.asciidoc [hunk 14]
```diff
@@ -455,7 +452,7 @@ See https://www.owasp.org/index.php/Information_exposure_through_query_strings_i
 [[config-global-labels]]
 ==== `GlobalLabels` (added[1.2])
 
-Labels added to all events, with the format `key=value[,key=value[,...]]`.
+These are labels that are added to all events with the format `key=value[,key=value[,...]]`.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:455]:
```suggestion
Labels are added to all events with the format `key=value[,key=value[,...]]`.
```

File: docs/configuration.asciidoc [hunk 15]
```diff
@@ -502,9 +499,9 @@ IMPORTANT: Use of `ServerUrls` is deprecated. Use `ServerUrl`.
 A string used to ensure that only your agents can send data to your APM server.
 
 Both the agents and the APM server have to be configured with the same secret token.
-Use this setting in case the APM Server requires a token (e.g. APM Server in Elastic Cloud).
+Use this setting if the APM Server requires a token (e.g. APM Server in Elastic Cloud).
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:502]:
```suggestion
Use this setting if the APM Server requires a secret token, for example, when using our hosted {es} Service on Elastic Cloud.
```

File: docs/configuration.asciidoc [hunk 16]
```diff
@@ -567,7 +564,7 @@ Verification can be disabled by changing this setting to false.
 [[config-server-cert]]
 ==== `ServerCert` (added[1.9])
 
-The path to a PEM-encoded certificate used for SSL/TLS by APM server, used to perform validation 
+The path to a PEM-encoded certificate used for SSL/TLS by APM server, is used to perform validation
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:567]:
```suggestion
The path to a PEM-encoded certificate used for SSL/TLS by APM server. Used to perform validation 
```

File: docs/configuration.asciidoc [hunk 17]
```diff
@@ -773,7 +769,7 @@ Possible options are `off`, `errors`, `transactions` and `all`:
 * `transactions` - request bodies will only be reported with request transactions
 * `all` - request bodies will be reported with both errors and request transactions
 
-This setting can be changed after agent's start.
+This setting can be changed after the agent is started.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:772]:
```suggestion
This setting can be changed after the agent starts.
```

File: docs/configuration.asciidoc [hunk 18]
```diff
@@ -791,13 +787,13 @@ This setting can be changed after agent's start.
 [[config-capture-body-content-types]]
 ==== `CaptureBodyContentTypes` (performance) (added[1.0.1])
 
-Configures which content types should be captured.
+This determines which content types should be captured.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:790]:
```suggestion
Configures the content types to be captured.
```

File: docs/configuration.asciidoc [hunk 19]
```diff
@@ -791,13 +787,13 @@ This setting can be changed after agent's start.
...
 
 This option supports the wildcard `*`, which matches zero or more characters.
 Examples: `/foo/*/bar/*/baz*`, `*foo*`.
 Matching is case insensitive.
 
-This setting can be changed after agent's start.
+This setting can be changed after the agent is started.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:796]:
```suggestion
This setting can be changed after the agent starts.
```

File: docs/configuration.asciidoc [hunk 20]
```diff
@@ -820,7 +816,7 @@ This setting can be changed after agent's start.
 If set to `true`,
 the agent will capture request and response headers, including cookies.
 
-NOTE: Setting this to `false` reduces memory allocations, network bandwidth and disk space used by Elasticsearch.
+NOTE: Setting this to `false` reduces memory allocations, network bandwidth, and disk space used by Elasticsearch.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:819]:
```suggestion
NOTE: Setting this to `false` reduces memory allocations, network bandwidth, and disk space used by {es}.
```

File: docs/configuration.asciidoc [hunk 21]
```diff
@@ -840,10 +836,10 @@ NOTE: Setting this to `false` reduces memory allocations, network bandwidth and
 
 <<dynamic-configuration, image:./images/dynamic-config.svg[] >>
 
-Used to restrict requests to certain URLs from being instrumented.
+This is used to restrict requests to certain URLs from being instrumented.
 
 This property should be set to a list containing one or more strings.
-When an incoming HTTP request is detected, its request path will be tested against each element in this list.
+When an incoming HTTP request is detected, the request path will be tested against each element in this list.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:842]:
```suggestion
When an incoming HTTP request is detected, its request path will be tested against each element in this list.
```

File: docs/configuration.asciidoc [hunk 22]
```diff
@@ -921,7 +917,7 @@ This suppresses any configuration of `ExcludedNamespaces`.
 [[config-excluded-namespaces]]
 ==== `ExcludedNamespaces` (added[1.5])
 
-A list of namespaces to exclude when reading an exception's StackTrace to determine the culprit.
+A list of namespaces to exclude when reading an exceptions StackTrace to determine the culprit.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:920]:
```suggestion
A list of namespaces to exclude when reading an exception StackTrace to determine the culprit.
```

File: docs/configuration.asciidoc [hunk 23]
```diff
@@ -942,7 +938,7 @@ Namespaces are checked with `string.StartsWith()`, so "System." matches all Syst
 
 <<dynamic-configuration, image:./images/dynamic-config.svg[] >>
 
-Setting it to 0 will disable stack trace collection. Any positive integer value will be used as the maximum number of frames to collect. Setting it to -1 means that all frames will be collected.
+Setting this to `0` disables the stack trace collection. Any positive integer value will be used as the maximum number of frames to collect. Setting it to -1 means that all frames will be collected.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:941]:
```suggestion
Setting this to `0` disables stack trace collection. Any positive integer value will be used as the maximum number of frames to collect. Setting it to -1 means that all frames will be collected.
```

**Comment by bmorelli25** [comment]
Jenkins, run the tests please

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Review by bmorelli25 — APPROVED** [review]
Pending doc build, LGTM

File: docs/configuration.asciidoc [hunk 24]
```diff
@@ -88,8 +88,8 @@ Here is a sample `appsettings.json` configuration file for a typical ASP.NET Cor
 <1> With ASP.NET Core, you must set `LogLevel` for the internal APM logger in the standard `Logging` section with the `Elastic.Apm` category name.
 <2> The configurations below `ElasticApm` are fetched by the agent if the corresponding `IConfiguration` is passed to the agent.
 
-In certain scenarios--like when you're not using ASP.NET Core--you won't activate the agent with the `UseElasticApm()` method.
-In this case, you can set the agent's log level with <<config-log-level,`ElasticApm:LogLevel`>>, as shown in the following `appsettings.json` file:
+In certain scenarios--you won't activate the agent with the `UseElasticApm()` method.
```
Comment (russcam) [inline docs/configuration.asciidoc:91]: stray `--` still present- maybe replace with comma?

**Review by russcam — APPROVED** [review]
LGTM, left one small comment

File: docs/configuration.asciidoc (hunk 24 above)
Comment (bmorelli25) [reply docs/configuration.asciidoc:91]: Ahh, dang, I hadn't refreshed the page so this comment didn't show up until after I pressed the merge button. Let me quick open a new PR to get this fixed 🙃

Comment (bmorelli25) [reply docs/configuration.asciidoc:91]: I'm not sure why the "like when you're not using ASP.NET Core" was removed here tbh. I think it makes sense to keep it.

**Pull request merged**